### DISCUSSION TITLE
How to insert script to head with app router in RSC?

### DISCUSSION THREAD

**Discussion by CaptainN** [vercel/next.js#66089] [App Router]
I'd like to insert a script tag to the head element, so that it is there in when the HTML loads, but I want to insert that from a layout group's layout (not the root layout). In a root layout, there is `next/document`'s head element, but there doesn't seem to be a corresponding element for RSC components (including layout group layouts). `next/head` would be obvious to reach for, but it doesn't work.

The docs seem to emphasize using direct export meta attributes, but those don't work for 100% of cases - they only really work for predefined use cases. That's not the most flexible tool. Is there a way to insert whatever I want there?

**Comment by icyJoseph** [comment]
Well, React supports this:

- https://react.dev/reference/react-dom/components/script#rendering-an-external-script

As long as you provide a `src` and `async` attributes, it will injected at the head.
